### DISCUSSION TITLE
Error about GLEW when configuring with CMake 3.25.0-rc4 gui

### DISCUSSION THREAD

**Discussion by Matyas37** [FLAMEGPU/FLAMEGPU2#989] [Q&A] [answered]
Hello,

I am trying to configure FlameGPU with visualisation using the CMake GUI. I have downloaded the binaries zip file from https://glew.sourceforge.net and extracted it. Then I ran CMake and got this error.

![image](https://user-images.githubusercontent.com/93601565/203058829-28233e08-ca3d-42eb-9224-9fa72fbfb61f.png)

My folders with FlameGPU and GLEW are both in C: on the same level.
![image](https://user-images.githubusercontent.com/93601565/203059053-79e2532e-3eb0-4ad1-94d1-54b295ad3e95.png)

I have included a few more screenshots of CMake in case it helps.
![image](https://user-images.githubusercontent.com/93601565/203059285-c315e06e-6585-471b-adad-d98d1f78b1f8.png)
![image](https://user-images.githubusercontent.com/93601565/203059468-e41731ba-3031-4e89-8799-444e63c8b816.png)
![image](https://user-images.githubusercontent.com/93601565/203059580-46b8566b-c237-4895-bdcd-940cf9b03e13.png)
![image](https://user-images.githubusercontent.com/93601565/203059638-db6c9675-aee4-44a6-b686-432007cfc795.png)

**Comment by ptheywood** [comment, marked as answer]
The [FLAMEGPU2-visualsier CMake](https://github.com/FLAMEGPU/FLAMEGPU2-visualiser/blob/master/cmake/dependencies/glew.cmake) is currently set up for Windows buildsfetch the pre-compiled GLEW from https://github.com/nigels-com/glew, and then find that version of GLEW.

However, it doesn't explicitly only check in the expected location however, nor only check for a config file, which might be allowing it to find the version you have downloaded and extracted. 

There are a few options:

1. Remove  the manually extracted version of GLEW, this *might* allow you to configure CMake, but its' not a true fix. 
2. We can improve the FLAMEGPU2-visualsier GLEW CMake:
  1. We can make it so we only try to find the version of GLEW CMake downlaods - (untested https://github.com/FLAMEGPU/FLAMEGPU2-visualiser/pull/109) 
  2. We can attempt to find a system-wide GLEW, and if it is found and is the correct version use that, otherwise fetch GLEW ourselves on windows. This is not as simple as it sounds unfortunatley given how `find_package` works in practice (on Windows).

---

Would you be able to try the patched version of our visualisation CMake I've put together? (I'm not on windows so can't attempt to reproduce your error): 

You *should* be able to do this by setting the value of `VISUALISATION_GIT_VERSION` to `0da56feada1bc3fe76eec241cdcb35da876e01d2` and running the CMake Generation. There is a chance the CMake caching of fetched resources might be a little agressive however (I can't remember changing this variable works after the initial configure or not on Windows), in which case you can either:

+ Clear the `build/_deps` derectory and reconfigure, again this might not work
+ Change Line 11 of `cmake/dependencies/flamegpu2-visualiser` to `0da56feada1bc3fe76eec241cdcb35da876e01d2`, clear the entire build directory (or potentially just `CMakeCache.txt`) and reconfigure
+ Open a terminal with `git` available, and an exising build dir configure attempt, navigate to `build/_deps/flamegpu2_visualiser-src` and run `get fetch && git checkout find-fetched-glew` to switch to the appropraite commit with the patch (or just apply the patch yourself)

If this fixes the problem, we can get it merged in and the visusalisation hash updated.

**Reply by Matyas37** [reply]
I replaced line 11 of  `cmake/dependencies/flamegpu2-visualiser` and then deleted the build directory and reconfigured and it worked! I was able to configure and run some of the example simulations.

Thank you.

**Reply by ptheywood** [reply]
Sweet, we'll get that PR cleaned up and merged in, so going forwards no patching will be required. 

Thanks for letting us know about the issue.
